File: 1. Bronze/Problem_1009.java
/*
* 백준 1009번. 분산처리.
* https://www.acmicpc.net/problem/1009
* */
import java.util.Scanner;
//Todo: Fix Error
public class Problem_1009 {
    public static void main(String[] args){
        int a;
        int b;
        int n; // test case

        Scanner s = new Scanner(System.in);
        n = s.nextInt();

        for(int i=0;i<n;i++){
            a = s.nextInt();
            b = s.nextInt();

            // 모든 숫자는 자신의 숫자를 여러 번 곱할 때 최대 4주기를 가진다.
            // N^4와 N^0은 다르므로 4를 추가
            b = b % 4 + 4;

            // 결과를 계산하고 저장
            // 0번 컴퓨터는 없다. % 했을 때 0이면 10으로 바꿔주자.
            int res = (int)Math.pow(a, b)%10;
            if(res == 0)
                System.out.println(10);
            else
                System.out.println(res);

        }

    }

}

File: 1. Bronze/Problem_1009.kt
import kotlin.math.pow

/*
* 백준 1009번. 분산처리
* https://www.acmicpc.net/problem/1009
*/

fun main() {

    val br = System.`in`.bufferedReader()
    val n = br.readLine().toInt()
    val result = arrayListOf<Int>()

    for (i in 0 until n) {

        var (a, b) = br.readLine().split(' ').map { it.toInt() }

        a %= 10 // 구하는 수는 일의 자리이므로 일의 자리만 신경 쓴다.
        b = b % 4 + 4 // 모든 수는 자기 자신을 곱했을 때 마지막 자리가 4번의 주기로 반복된다.

        val computer = a.toDouble().pow(b.toDouble()).toInt() % 10

        if (computer == 0) result.add(10)
        else result.add(computer)
    }

    val bw = System.out.bufferedWriter()
    result.forEach {
        bw.write("$it\n")
    }
    bw.flush()
    bw.close()
}
